src/algebra/module/basic.lean

```diff
@@ -487,3 +487,39 @@ by rw [nsmul_eq_mul, mul_one]
 @[simp] lemma int.smul_one_eq_coe {R : Type*} [ring R] (m : ℤ) :
   m • (1 : R) = ↑m :=
 by rw [gsmul_eq_mul, mul_one]
+
+namespace nat
+
+variables {A : Type*} [add_comm_monoid A]
+
+/-- `nat.cast_add_hom' a` is the additive monoid homomorphism `ℕ → A`
+that sends `1 : ℕ` to `a : A`. -/
+def cast_add_hom' (a : A) : ℕ →+ A :=
+{ to_fun := λ n, n • a,
+  map_zero' := zero_smul _ _,
+  map_add' := λ m n, add_smul m n a }
```

Comment:
Isn't this `smul_add_hom` or similar?

Reply:
Good point. Turns out you also need a `flip`. Not sure if these are worth keeping, or that I should just use the `(smul_add_hom _ _).flip` directly.

Reply:
I think they're probably not worth keeping as is. If this comes up often you could probably come up with a nicer name for `(smul_add_hom R M).flip`, but I don't think the special cases for `int` and `nat` are useful.